internal/configs/configmaps_test.go

```diff
@@ -285,6 +285,168 @@ func TestParseConfigMapAccessLogDefault(t *testing.T) {
+					"license-token-secret-name": "license-token",
+					"enforce-initial-report":    "7",
+				},
+			},
+			msg: "enforce-initial-report is invalid",
+		},
+		{
+			configMap: &v1.ConfigMap{
+				Data: map[string]string{
+					"license-token-secret-name": "license-token",
+					"enforce-initial-report":    "",
+				},
+			},
+			msg: "enforce-initial-report set empty",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.msg, func(t *testing.T) {
+			_, configWarnings, _ := ParseMGMTConfigMap(context.Background(), test.configMap, makeEventLogger())
+
+			if !configWarnings {
+				t.Fatal("Expected warnings, got none")
+			}
+		})
+	}
+}
+
+func TestParseMGMTConfigMapLicense(t *testing.T) {
+	t.Parallel()
+	test := struct {
+		configMap *v1.ConfigMap
+		want      *MGMTConfigParams
+		msg       string
+	}{
+		configMap: &v1.ConfigMap{
+			Data: map[string]string{
+				"license-token-secret-name": "license-token",
+			},
+		},
+		want: &MGMTConfigParams{
+			Secrets: MGMTSecrets{
+				License: "license-token",
+			},
+		},
+		msg: "Has only license-token-secret-name",
+	}
+
+	t.Run(test.msg, func(t *testing.T) {
+		result, warnings, err := ParseMGMTConfigMap(context.Background(), test.configMap, makeEventLogger())
+		if err != nil {
+			t.Fatal(err)
+		}
+		if warnings {
+			t.Fatal("Unexpected warnings")
+		}
+		if result.Secrets.License != test.want.Secrets.License {
+			t.Errorf("LicenseTokenSecretNane: want %q, got %q", test.want.Secrets.License, result.Secrets.License)
+		}
+	})
+}
+
+func TestParseMGMTConfigMapEnforceInitialReport(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		configMap *v1.ConfigMap
+		want      *MGMTConfigParams
+		msg       string
+	}{
+		{
+			configMap: &v1.ConfigMap{
+				Data: map[string]string{
+					"license-token-secret-name": "license-token",
+					"enforce-initial-report":    "false",
+				},
+			},
+			want: &MGMTConfigParams{
+				EnforceInitialReport: BoolToPointerBool(false),
+				Secrets: MGMTSecrets{
+					License: "license-token",
+				},
+			},
+			msg: "enforce-initial-report set to false",
+		},
+		{
+			configMap: &v1.ConfigMap{
+				Data: map[string]string{
+					"license-token-secret-name": "license-token",
+					"enforce-initial-report":    "true",
+				},
+			},
+			want: &MGMTConfigParams{
+				EnforceInitialReport: BoolToPointerBool(true),
+				Secrets: MGMTSecrets{
+					License: "license-token",
+				},
+			},
+			msg: "enforce-initial-report set to true",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.msg, func(t *testing.T) {
+			result, warnings, err := ParseMGMTConfigMap(context.Background(), test.configMap, makeEventLogger())
+			if err != nil {
+				t.Fatal(err)
+			}
+			if warnings {
+				t.Fatal("Unexpected warnings")
+			}
+
+			if result.EnforceInitialReport == nil {
+				t.Fatalf("EnforceInitialReport: want %v, got nil", *test.want.EnforceInitialReport)
+			} else if *result.EnforceInitialReport != *test.want.EnforceInitialReport {
```

Comment:
`t.Fatal` will finish execution in case of error. Then `else` is obsolete.

Reply:
thanks, I deleted the else and added another if block